src/main/java/org/truffleruby/Layouts.java

```diff
@@ -101,6 +101,14 @@
 
 public abstract class Layouts {
 
+    // Standard identifiers
+    // These must appear before the generated layout list so the identifiers have been initialized by the time
+    // the layout singletons are created.
+
+    public static final HiddenKey OBJECT_ID_IDENTIFIER = new HiddenKey("object_id");
```

Comment:
Okay. I'll back it out. We can always add it again.

The caution with doing it globally is for an arbitrary class it could force overflowing into the extension array.